pandas/tests/arithmetic/test_timedelta64.py

```diff
@@ -59,6 +89,59 @@ def get_expected_name(box, names):
     return exname
 
 
+def get_result_type(
+    td64_type: Type[TD64_TYPE],
+    dt64_type: Type[DT64_TYPE],
+) -> Type[DT64_TYPE]:
+    """
+    Expected result when adding, subtracting timedelta64-valued box and
+    datetime64-valued box or scalar.
+    """
+    dt64 = tm.wrap_value(Timestamp.now(), dt64_type)
+    td64 = tm.wrap_value(Timedelta(0), td64_type)
+    return type(dt64 + td64)
```

Comment:
so this is going to map Timedelta->Timestamp and TimedeltaArray->DatetimeArray but be an identity mapping for everything else?

Reply:
I might be misunderstanding the question, but think the gist of what you're suggesting is right (assuming Array stands for every box type). So:

```
Timedelta + Timestamp -> Timestamp
Timedelta + DatetimeArray -> DatetimeArray
Timedelta + DatetimeIndex -> DatetimeIndex
Timedelta + Series -> Series
Timedelta + DataFrame -> DataFrame
TimedeltaArray + Timestamp -> DatetimeArray
TimedeltaArray + DatetimeArray -> DatetimeArray
...
TimedeltaIndex + Timestamp -> DatetimeIndex
TimedeltaIndex + DatetimeArray -> DatetimeIndex
...
```

Happy to update the docstring, or use a different pattern, if that's unclear.

Reply:
My preference (mentioned https://github.com/pandas-dev/pandas/pull/46854#issuecomment-1110507748) is to handle the Timedelta/Timestamp cases in the existing tests/scalar/... files, and the non-scalar tests here.  Then this wrapping becomes unnecessary and you can just use box_with_array

Reply:
How about scalar/box tests, e.g. `Timedelta + DatetimeIndex`, where should those live?